katello/katello/hostname-change.rb

```diff
@@ -369,18 +377,26 @@ def run
       installer << " --disable-system-checks" if disable_system_check_option?
 
       STDOUT.puts installer
-      installer_output = self.run_cmd("#{installer}")
-      installer_success = $?.success?
-      STDOUT.puts installer_output
-
-      STDOUT.puts "Restarting puppet services"
-      self.run_cmd("/sbin/service puppet restart")
-      self.run_cmd("katello-service restart --only puppetserver")
+      default_installer << ' --disable-system-checks' if disable_system_check_option?
+      failure_message = "Something went wrong with the #{@options[:scenario].capitalize} installer. Please check the above output and the corresponding logs. Once fixed you may complete the hostname change by running \"#{default_installer}\""
+      run_cmd(installer, [0], failure_message) do |result, success|
+        if temp_dir
+          if success
+            STDOUT.puts 'cleaning up temporary files'
+            FileUtils.remove_entry(temp_dir)
+          else
+            STDOUT.puts 'restoring last_scenario.yaml'
+            run_cmd("cp #{temp_dir}/last_scenario.yaml #{last_scenario_yaml}")
+          end
+        end
 
-      if installer_success
-        self.successful_hostname_change_message
-      else
-        self.fail_with_message("Something went wrong with the #{@options[:scenario].capitalize} installer! Please check the above output and the corresponding logs")
+        if success
+          STDOUT.puts result
+          STDOUT.puts 'Restarting puppet services'
+          run_cmd('/sbin/service puppet restart')
```

Comment:
service is old style. systemctl is the current way to go. Does EL7 already support the ```try-reload-or-restart``` since I think that's what we want here: prefer reload but otherwise restart only if it's running.

Reply:
ya we can do systemctl here, `/sbin/service` is leftover from RHEL6 support. I'm fine with just using restart, but don't know much about this `try-reload-or-restart` option (sound great if its supported)